src/server/auth/microsoft.ts

```diff
@@ -0,0 +1,34 @@
+import { OIDCStrategy as Strategy, IProfile } from 'passport-azure-ad';
+import { verifyCallback } from './helpers';
+import { Models } from '../models';
+
+const { MSFT_CLIENT_ID, MSFT_CLIENT_SECRET, MSFT_REDIRECT_URL, MSFT_TENANT_NAME } = process.env;
+
+if (!MSFT_CLIENT_ID) throw new Error('MSFT_CLIENT_ID not set');
+
+if (!MSFT_CLIENT_SECRET) throw new Error('MSFT_CLIENT_SECRET not set');
+
+if (!MSFT_REDIRECT_URL) throw new Error('MSFT_REDIRECT_URL not set');
+
+if (!MSFT_TENANT_NAME) throw new Error('MSFT_TENANT_NAME not set');
+
+export const strategy = (models: Models): Strategy =>
+	new Strategy(
+		{
+			allowHttpForRedirectUrl: true,
+			clientID: MSFT_CLIENT_ID,
+			clientSecret: MSFT_CLIENT_SECRET,
+			identityMetadata: MSFT_TENANT_NAME,
+			loggingLevel: 'info',
+			passReqToCallback: false,
+			redirectUrl: `http://localhost:8081${MSFT_REDIRECT_URL}`, // TODO: FIX
```

Comment:
Is there a good way for me to have the env var be `MSFT_REDIRECT_URL="/api/auth/microsoft/callback"` and still make the path relative to the deploy context?
`

Reply:
GitHub doesn't mind a relative redirect path, but Google doesn't allow it, so mostly I've just been setting the full path in the env var

Reply:
Gotcha. Also, how aggressive should the loggingLevel be - `info`, `warn` or `error` (or none). And should I log PII?

Reply:
info if no err, error if err, warn has no use case